wasm.md

```diff
@@ -1531,6 +1496,238 @@ Start Function
+           <exports>    ... NAME |-> TFIDX                        ... </exports>
+           ...
+         </moduleInst>
+         <memInst>
+           <mAddr> ADDR </mAddr>
+           <mmax>  MAX  </mmax>
+           <msize> SIZE </msize>
+           ...
+         </memInst>
+       requires #limitsMatchImport(SIZE, MAX, LIM)
+
+    rule <k> ( import MOD NAME (global OID:OptionalId TGTYP:TextGlobalType)) => . ... </k>
+         <curModIdx> CUR </curModIdx>
+         <moduleInst>
+           <modIdx> CUR </modIdx>
+           <globIds> IDS => #saveId(IDS, OID, NEXT) </globIds>
+           <globalIndices> GS => GS [NEXT <- ADDR] </globalIndices>
+           <nextGlobIdx> NEXT => NEXT +Int 1 </nextGlobIdx>
+           ...
+         </moduleInst>
+         <moduleRegistry> ... MOD |-> MODIDX ... </moduleRegistry>
+         <moduleInst>
+           <modIdx> MODIDX </modIdx>
+           <globIds> IDS' </globIds>
+           <globalIndices> ... #ContextLookup(IDS' , TFIDX) |-> ADDR ... </globalIndices>
+           <exports>     ... NAME |-> TFIDX                          ... </exports>
+           ...
+         </moduleInst>
+         <globalInst>
+           <gAddr>  ADDR    </gAddr>
+           <gValue> <TYP> _ </gValue>
+           <gMut>   MUT     </gMut>
+         </globalInst>
+       requires asGMut(TGTYP) ==K MUT TYP
+```
+
+Tables and memories have proper subtyping, unlike globals and functions where a type is only a subtype of itself.
+Subtyping is determined by whether the limits of one table/memory fit in the limits of another.
+The following function checks if the limits in the first parameter *match*, i.e. is a subtype of, the limits in the second.
+
+```k
+    syntax Bool ::= #limitsMatchImport(Int, OptionalInt, Limits) [function]
+ // -----------------------------------------------------------------------
+    rule #limitsMatchImport(L1,      _, L2:Int   ) => L1 >=Int L2
+    rule #limitsMatchImport( _,   .Int,  _:Int  _) => false
+    rule #limitsMatchImport(L1, U1:Int, L2:Int U2) => L1 >=Int L2 andBool U1 <=Int U2
+```
+
+Imports can also be declared like regular functions, memories, etc., by giving an inline import declaration.
+
+```k
+    syntax InlineImport ::= "(" "import" String String ")"
+ // ------------------------------------------------------
+
+    syntax GlobalSpec ::= InlineImport TextGlobalType
+ // -------------------------------------------------
+    rule <k> ( global OID:OptionalId (import MOD NAME) TYP ) => ( import MOD NAME (global OID TYP) ) ... </k>
+
+    syntax FuncSpec ::= InlineImport TypeUse
+ // ----------------------------------------
+    rule <k> ( func OID:OptionalId (import MOD NAME) TUSE ) => ( import MOD NAME (func OID TUSE) ) ... </k>
+
+    syntax TableSpec ::= InlineImport TableType
+ // -------------------------------------------
+    rule <k> ( table OID:OptionalId (import MOD NAME) TT:TableType ) => (import MOD NAME (table OID TT)) ... </k>
```

Comment:
syntax of `(import MOD NAME (table OID TT))` is different with `( import MOD NAME (func OID TUSE) )` (whether to have spaces)

Reply:
Fixed